### DISCUSSION TITLE
关于参考文献格式

### DISCUSSION THREAD

**Discussion by moyu3585** [nju-lug/NJUThesis#174] [Q&A] [answered]
想请教一个问题，我想设置参考文献期刊名卷号加粗，类似如图
![image](https://user-images.githubusercontent.com/39209867/217050188-8f23d0ac-2ab8-4dd2-9320-d5248e39acff.png)
查询了下biblatex GB7714-2015，发现\DeclareFieldFormat[article,periodical]{volume}{\textbf{#1}}
\DeclareFieldFormat[article,periodical]{number}{\textbf{#1}}类似命令可以，然而在南大tex导言区加这个使用报错

**Comment by atxy-blip** [comment, marked as answer]
`njuthesis` 为了吸收导言区的用户设置，将 `biblatex` 挂在导言区结束后载入。所以你需要利用钩子机制来把这些设置放到参考文献宏包载入后，也就是在导言区加入这样一段：

```tex
\usepackage{etoolbox}
\AtEndPreamble{
  \DeclareFieldFormat[article,periodical]{volume}{\textbf{#1}}
  \DeclareFieldFormat[article,periodical]{number}{\textbf{#1}}
}
```
